cpp/modmesh/transform/fourier.hpp

```diff
@@ -63,22 +67,37 @@ class FourierTransform
     FourierTransform & operator=(FourierTransform && other) = delete;
 
     template <template <typename> class T1, typename T2>
-    static void fft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
+    static void fft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out, std::string const && backend)
     {
         const size_t N = in.size();
 
-        if ((N & (N - 1)) == 0)
+        if (backend == "cpu")
+        {
+            if ((N & (N - 1)) == 0)
+            {
+                detail::fft_radix_2<T1, T2>(in, out);
+            }
+            else
+            {
+                detail::fft_bluestein<T1, T2>(in, out);
+            }
+        }
+        else if (backend == "cuda")
         {
-            detail::fft_radix_2<T1, T2>(in, out);
+#if defined(BUILD_CUDA)
+            modmesh::fft_cuda<T1, T2>(in, out);
+#else
+            throw std::runtime_error("CUDA is not available.");
+#endif
         }
         else
         {
-            detail::fft_bluestein<T1, T2>(in, out);
+            throw std::runtime_error("unsupported backend.");
         }
     }
 
     template <template <typename> class T1, typename T2>
-    static void ifft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
+    static void ifft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out, std::string const && backend)
```

Comment:
Use an `enum` for `backend`.  Do not use a string which can only be checked during runtime.